### ISSUE TITLE
Del operator false negative unbound variable

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#3795]
**Describe the bug**
Apply del operator to a variable in function scope does not catch unbound variable.

```python
del x # Good: Error reported

def foo():
    del y # No error reported

class Foo:
    del z # Good error reported.
```
 
**Expected behavior**
I would expect all 3 del lines to report an error. All of them crash.

**VS Code extension or command-line**
pyright  cli 1.1.265.

**Additional context**
This looks like a regression from 1.1.217 -> 1.1.218.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.266, which I just published. It will also be included in a future release of pylance.
